src/features/defi/providers/fox-farming/components/FoxFarmingManager/Claim/ClaimStatus.tsx

```diff
@@ -128,20 +137,43 @@ export const ClaimStatus: React.FC<ClaimStatusProps> = ({ accountId }) => {
   }, [txid, accountAddress, accountId])
   const confirmedTransaction = useAppSelector(gs => selectTxById(gs, serializedTxIndex))
 
+  const status = useMemo(() => {
+    if (maybeSafeTx?.isQueuedSafeTx) {
+      return TxStatus.Pending
+    }
+
+    if (maybeSafeTx?.isExecutedSafeTx) {
+      return TxStatus.Confirmed
+    }
+
```

Comment:
Gut feel is having SAFE data for *all* Txs vs. lazily may not be a good idea, though we early bail on sc checks so it may or may not be safe, will take a look!
Conceptually like the idea minus the eager vs. lazy ramifications.

Reply:
Regarding the amount of work, we shouldn't include this in this PR anyway and could be done later in a follow up if we choose